.github/workflows/release.yml

```diff
@@ -3,24 +3,57 @@ name: Release on Pypi and GitHub
 on:
   workflow_dispatch:
 
+  push:
+    tags:
+      - 'v*'  # Trigger on version tags like v1.0.0
```

Comment:
Original workflow assumed you are gonna initiate this workflow manually and as part of it both tag and GH release is created. What is expected release workflow now?

Reply:
The workflow is to tag a version on master branch, and this flow would be triggered to do the release.

Version of the tag is what would be used to publish

Hence you can release with commiting any extra code related to versioning

Reply:
yes, the git tag would be the source of truth